time/unit.go

```diff
@@ -56,6 +56,14 @@ func (tu Unit) Value() (time.Duration, error) {
 	return 0, errUnrecognizedTimeUnit
 }
 
+// Count returns the number of units contained within the duration.
+func (tu Unit) Count(d time.Duration) (int, error) {
+	if dur, found := unitsToDuration[tu]; found {
+		return int(d / dur), nil
+	}
+	return 0, errUnrecognizedTimeUnit
```

Comment:
Feels weird to have a method on `Unit` return an error on unrecognised units. Maybe push this on the user, where they have to guarantee the Unit IsValid before calling this, can do an internal check in this method and return `-1` if invalid.